core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java

```diff
@@ -129,10 +129,18 @@ private void throwBadRequest(StackView stack) {
                 stack.getName(), stack.getStatus()));
     }
 
-    public void validateStackStatusForStartHostGroup(StackView stack) {
-        if (!(stack.isAvailable() || stack.isAvailableWithStoppedInstances())) {
-            throw new BadRequestException(format("Data Hub '%s' has '%s' state. Node group start operation is not allowed for this state.",
-                    stack.getName(), stack.getStatus()));
+    public void validateStackStatusForStartHostGroup(StackDto stack) {
+        CmTemplateProcessor processor = cmTemplateProcessorFactory.get(stack.getBlueprint().getBlueprintText());
+        List<String> unhealthyHostsGroupNames = processor.getUnhealthyDependentHostGroups(stack);
+        if (unhealthyHostsGroupNames.size() > 0) {
+            if (unhealthyHostsGroupNames.get(0).equals("No check required")) {
+                if (!(stack.isAvailable() || stack.getStack().isAvailableWithStoppedInstances())) {
+                    throw new BadRequestException(format("Data Hub '%s' has '%s' state. Node group start operation is not allowed for this state.",
+                            stack.getStack().getName(), stack.getStack().getStatus()));
+                }
+            } else {
+                throw new BadRequestException(format("Upscaling is Not Allowed as Data hub '%s' has Unhealthy Nodes ", stack.getStack().getName()));
+            }
```

Comment:
"No check required" will be removed if `CMRepositoryVersionUtil` is used. Any way of simplifying this if else ladder?

Reply:
Yes, "No Check required" will be removed and the if else ladder will also get simplified.

Reply:
Done